src/main/groovy/org/spongepowered/asm/gradle/plugins/MixinExtension.groovy

```diff
@@ -523,7 +527,7 @@ public class MixinExtension {
     @PackageScope Set<SourceSet> findMissingAnnotationProcessors() {
         Set<SourceSet> missingAPs = []
         missingAPs += this.sourceSets.findResults { SourceSet sourceSet ->
-            sourceSet.ext.mixinDependency = this.findMixinDependency(sourceSet.compileConfigurationName) ?: this.findMixinDependency(sourceSet.implementationConfigurationName)
+            sourceSet.ext.mixinDependency = this.findMixinDependency(sourceSet.implementationConfigurationName)
```

Comment:
This also could probably be better addressed by letting the users specify the config name instead of guessing.

Reply:
This method is only for throwing an exception if the user doesn't have an Annotation Processor specified but does have a mixin dependency. I don't think it is necessary to let the user configure something that is purely used to throw an exception if they do something else wrong.

Reply:
Then support both G6- and G7+ You can detect what gradle version is running and pick which config to use.
Of you could go the other route and detect which config has mixin defined, and make sure there is a AP in there.
Either way, don't break compat for no reason.